installscripts/cookbooks/jenkins/recipes/sonarqube-scanner.rb

```diff
@@ -0,0 +1,53 @@
+remote_file "#{Chef::Config['file_cache_path']}/sonar-scanner-cli-3.0.3.778-linux.zip" do
+  source 'https://sonarsource.bintray.com/Distribution/sonar-scanner-cli/sonar-scanner-cli-3.0.3.778-linux.zip'
+  mode '0755'
+  action :create
+end
+
+execute 'install_sonarscanner' do
+  command "unzip -o #{Chef::Config['file_cache_path']}/sonar-scanner-cli-3.0.3.778-linux.zip"
+  cwd '/opt'
+end
+
+directory '/opt/sonar-scanner-3.0.3.778-linux/' do
+  owner 'jenkins'
+  group 'jenkins'
+  recursive true
+end
+
+link '/opt/sonar-scanner-3.0.3.778-linux/bin/sonar-scanner' do
```

Comment:
@bleggett Not sure, but symlinks are not created. I had also tried this earlier but could not found out the reason. So I ended up with this,

> execute 'createsymlink' do
  command 'sudo ln -sf /opt/sonar-scanner-3.0.3.778-linux/bin/sonar-scanner /bin/sonar-scanner'
  ignore_failure true
end

Reply:
This is the "correct" way to do it, I will try to figure out why it doesn't work in our case.

Reply:
@rajeevr2715 I misread the Chef docs.

```
link '/opt/dependency-check/bin/dependency-check.sh' do
  to '/bin/dependency-check.sh'
end
```

actually tries to create a softlink NAMED `/opt/dependency-check/bin/dependency-check.sh` that links back TO `/bin/dependency-check.sh`, which is definitely not what we want.

Oddly enough this fails silently, no "file not found" errors or anything, unless you try to create a hardlink instead.

This is what it should be, and this works.
```
link '/bin/dependency-check.sh' do
  to '/opt/dependency-check/bin/dependency-check.sh'
end
```

This misunderstanding has been made by me in a few other spots, will fix.